# RenoStrategix: Transforming Homes, Elevating Values.

# <img src="kings.PNG" alt="RenoStrategix: Transforming Homes, Elevating Values" length ="100" width="1100">

## Authors.
* Caroline Njeri.
* Amadi Growman.
* Lynns Waswa.
* Robert Gasembe.
* David Kirianja.
* James Nyamu.


## Business Overview.

Welcome to RenoStrategix, where the world of real estate meets the art and science of optimizing home renovations for increased property value. In this project, we address a real-world problem faced by our visionary real estate agency known as "CasaCrafters Realty Solutions."
CasaCrafters aims to revolutionize the way homeowners approach renovations by providing data-driven advice on how specific home improvements impact property values. The primary challenge is to assist homeowners in making informed decisions about renovations that yield the highest return on investment. To achieve this objective, we will use multiple linear regression modeling to analyze house sales data in the King County area. Our focus is not just on increasing value but also on enhancing the overall appeal for potential buyers.




## Business Problem

Homeowners often find themselves at a crossroads when deciding on renovations. The lack of clear guidance on which improvements will significantly enhance their property values becomes a real-world challenge. CasaCrafters seeks to bridge this gap by offering tailored advice based on a comprehensive analysis of the King County real estate market.

In the vibrant landscape of King County, CasaCrafters endeavors to create a narrative that extends beyond buying and selling homes. Imagine a homeowner named Alex, who dreams of transforming their house into a haven. However, Alex is uncertain about which renovations will not only fulfill personal desires but also enhance the property's market value.

Through the journey of Alex's home transformation, CasaCrafters navigates the complex realm of real estate, uncovering insights that go beyond the expected. The story unfolds as Alex learns about the potential return on investment for various renovations, transforming not just the home but the overall real estate experience.


# Objective 1: Identify Key Features Impacting Property Values

**Objective:** Develop a multiple linear regression model to identify and quantify the influence of various features (e.g., square footage, number of bedrooms, location) on the sale prices of houses in King County.

**Rationale:** Understanding the key features that significantly affect property values is crucial for providing targeted recommendations to homeowners. By analyzing historical sales data, the model will reveal which features have the most substantial impact on sale prices. This information will empower CasaCrafters to guide homeowners on prioritizing renovations that are likely to yield the highest returns.

---

# Objective 2: Build a Predictive Model for Property Valuation

**Objective:** Construct a robust multiple linear regression model that accurately predicts house sale prices based on selected features, allowing for personalized property valuation.

**Rationale:** The predictive model will serve as a valuable tool for CasaCrafters to estimate the potential impact of specific renovations on a property's value. By inputting proposed changes into the model, homeowners can receive personalized predictions of how these renovations might affect the sale price. This enables informed decision-making and helps homeowners focus on improvements that align with their goals while maximizing return on investment.

---

# Objective 3: Provide Renovation Recommendations for Maximum ROI

**Objective:** Utilize the developed multiple linear regression model to generate personalized recommendations for homeowners, suggesting specific renovations that are predicted to have the highest positive impact on property values.

**Rationale:** CasaCrafters aims to be a trusted advisor for homeowners seeking to enhance their properties. By leveraging the regression model's insights, the agency can offer tailored recommendations, outlining the renovations that are not only aligned with the homeowner's vision but also expected to yield the greatest return on investment. This proactive approach adds significant value to CasaCrafters' services, fostering trust and satisfaction among homeowners.

## Data Understanding.
This project uses the King County House Sales dataset, which can be found in kc_house_data.csv in the data folder in this assignment's GitHub repository. The description of the column names can be found in column_names.md in the same folder. As with most real world data sets, the column names are not perfectly described, so you'll have to do some research or use your best judgment if you have questions about what the data means.

It is up to you to decide what data from this dataset to use and how to use it. If you are feeling overwhelmed or behind, we recommend you ignore some or all of the following features:

* date
* view
* sqft_above
* sqft_basement
* yr_renovated
* zipcode
* lat
* long
* sqft_living15
* sqft_lot15

#Data Pre-Processing#

The first task to read the data file into our working environment then explore it to gain an initial
understanding of the dataset. Moreover, this step also allows us to determine what data wrangling techniques to apply in order to
transform the data into a form that can be analysed and modeled.

In [34]:
# Import the necessary libraries for the project

import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import statsmodels.api as sm
import scipy.stats as stats

# from matplotlib import style    # NB: We can determine which style to use
# style.use('ggplot')
# style.use('seaborn-v0_8-whitegrid')

%matplotlib inline

In [35]:
# Load the file by using a file path into the notebook's working memory
import os

file_path = os.path.join('data', 'kc_house_data.csv')
data = pd.read_csv(file_path)



#Exploratory data analysis

In [36]:
# Check the shape of the dataset
data.shape

(21597, 21)

In [37]:
# Inspect the properties of the dataset

data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21597 entries, 0 to 21596
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21597 non-null  int64  
 1   date           21597 non-null  object 
 2   price          21597 non-null  float64
 3   bedrooms       21597 non-null  int64  
 4   bathrooms      21597 non-null  float64
 5   sqft_living    21597 non-null  int64  
 6   sqft_lot       21597 non-null  int64  
 7   floors         21597 non-null  float64
 8   waterfront     19221 non-null  object 
 9   view           21534 non-null  object 
 10  condition      21597 non-null  object 
 11  grade          21597 non-null  object 
 12  sqft_above     21597 non-null  int64  
 13  sqft_basement  21597 non-null  object 
 14  yr_built       21597 non-null  int64  
 15  yr_renovated   17755 non-null  float64
 16  zipcode        21597 non-null  int64  
 17  lat            21597 non-null  float64
 18  long  

In [38]:
# Dataset description
data.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.159700e+04  2.159700e+04  21597.000000  21597.000000  21597.000000   
mean   4.580474e+09  5.402966e+05      3.373200      2.115826   2080.321850   
std    2.876736e+09  3.673681e+05      0.926299      0.768984    918.106125   
min    1.000102e+06  7.800000e+04      1.000000      0.500000    370.000000   
25%    2.123049e+09  3.220000e+05      3.000000      1.750000   1430.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    sqft_above      yr_built  yr_renovated  \
count  2.159700e+04  21597.000000  21597.000000  21597.000000  17755.000000   
mean   1.509941e+04      1.494096   1788.596842   1970.999676     83.636778   
std    4.141264e+04      0.539683    827.759761     29.375234    399.946414   
min    5.200000e+02      1.000000    370.000000   1900.000000      0.000000   
25%    5.040000e+03      1.000000   1190.000000   1951.000000      0.000000   
50%    7.618000e+03      1.500000   1560.000000   1975.000000      0.000000   
75%    1.068500e+04      2.000000   2210.000000   1997.000000      0.000000   
max    1.651359e+06      3.500000   9410.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21597.000000  21597.000000  21597.000000   21597.000000   21597.000000  
mean   98077.951845     47.560093   -122.213982    1986.620318   12758.283512  
std       53.513072      0.138552      0.140724     685.230472   27274.441950  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471100   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.231000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [39]:
# Sample the dataset

data.head()

id        date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  10/13/2014  221900.0         3       1.00         1180   
1  6414100192   12/9/2014  538000.0         3       2.25         2570   
2  5631500400   2/25/2015  180000.0         2       1.00          770   
3  2487200875   12/9/2014  604000.0         4       3.00         1960   
4  1954400510   2/18/2015  510000.0         3       2.00         1680   

   sqft_lot  floors waterfront  view  ...          grade sqft_above  \
0      5650     1.0        NaN  NONE  ...      7 Average       1180   
1      7242     2.0         NO  NONE  ...      7 Average       2170   
2     10000     1.0         NO  NONE  ...  6 Low Average        770   
3      5000     1.0         NO  NONE  ...      7 Average       1050   
4      8080     1.0         NO  NONE  ...         8 Good       1680   

   sqft_basement yr_built  yr_renovated  zipcode      lat     long  \
0            0.0     1955           0.0    98178  47.5112 -122.257   
1          400.0     1951        1991.0    98125  47.7210 -122.319   
2            0.0     1933           NaN    98028  47.7379 -122.233   
3          910.0     1965           0.0    98136  47.5208 -122.393   
4            0.0     1987           0.0    98074  47.6168 -122.045   

   sqft_living15  sqft_lot15  
0           1340        5650  
1           1690        7639  
2           2720        8062  
3           1360        5000  
4           1800        7503  

[5 rows x 21 columns]

##Data cleaning

In [40]:
#Checking for null values
pd.DataFrame(data.isna().sum()).T

id  date  price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   0     0      0         0          0            0         0       0   

   waterfront  view  ...  grade  sqft_above  sqft_basement  yr_built  \
0        2376    63  ...      0           0              0         0   

   yr_renovated  zipcode  lat  long  sqft_living15  sqft_lot15  
0          3842        0    0     0              0           0  

[1 rows x 21 columns]

In [41]:
# Replacing the Null values at Waterfront with "No"
data['waterfront'].fillna('NO', inplace=True)

In [42]:
# Replacing the null values in the year renovated with the
# values in the year built
data['yr_renovated'].fillna(data['yr_built'], inplace=True)

The assumption made in this case was that the datapoints where there was a NaN value represented houses that had never been renovated.

In [43]:
# Dropping the null values in the view column
data.dropna(inplace=True)

In [44]:
# Replacing "?" with 0 in sqft_basement column
data['sqft_basement'] = data['sqft_basement'].replace('?', 'None', inplace=True)

The assumption made in this case was that the datapoints where there was a '?' represented houses that have no basement.

In [45]:
# Checking for null values after cleaning the data
pd.DataFrame(data.isna().sum()).T

id  date  price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0   0     0      0         0          0            0         0       0   

   waterfront  view  ...  grade  sqft_above  sqft_basement  yr_built  \
0           0     0  ...      0           0          21534         0   

   yr_renovated  zipcode  lat  long  sqft_living15  sqft_lot15  
0             0        0    0     0              0           0  

[1 rows x 21 columns]

In [46]:
data.info

<bound method DataFrame.info of                id        date     price  bedrooms  bathrooms  sqft_living  \
0      7129300520  10/13/2014  221900.0         3       1.00         1180   
1      6414100192   12/9/2014  538000.0         3       2.25         2570   
2      5631500400   2/25/2015  180000.0         2       1.00          770   
3      2487200875   12/9/2014  604000.0         4       3.00         1960   
4      1954400510   2/18/2015  510000.0         3       2.00         1680   
...           ...         ...       ...       ...        ...          ...   
21592   263000018   5/21/2014  360000.0         3       2.50         1530   
21593  6600060120   2/23/2015  400000.0         4       2.50         2310   
21594  1523300141   6/23/2014  402101.0         2       0.75         1020   
21595   291310100   1/16/2015  400000.0         3       2.50         1600   
21596  1523300157  10/15/2014  325000.0         2       0.75         1020   

       sqft_lot  floors waterfront  view  .

In [47]:
# Creating a new column to find the difference between the year renovated and year built
# 0 Has been replaced for the rows that returned -ve years.
data['Years_Since_Renovation'] = data.apply(lambda row: row['yr_renovated'] - row['yr_built'] if row['yr_renovated'] > 0 else 0, axis=1)


In [48]:
# Checking for any negative values in the new column.
if (data['Years_Since_Renovation'] < 0).any():
    print("There are negative values in the 'Years_Since_Renovation' column.")
else:
    print("There are no negative values in the 'Years_Since_Renovation' column.")


There are no negative values in the 'Years_Since_Renovation' column.


In [49]:
#Checking the data types in the data frame
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21534 entries, 0 to 21596
Data columns (total 22 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   id                      21534 non-null  int64  
 1   date                    21534 non-null  object 
 2   price                   21534 non-null  float64
 3   bedrooms                21534 non-null  int64  
 4   bathrooms               21534 non-null  float64
 5   sqft_living             21534 non-null  int64  
 6   sqft_lot                21534 non-null  int64  
 7   floors                  21534 non-null  float64
 8   waterfront              21534 non-null  object 
 9   view                    21534 non-null  object 
 10  condition               21534 non-null  object 
 11  grade                   21534 non-null  object 
 12  sqft_above              21534 non-null  int64  
 13  sqft_basement           0 non-null      object 
 14  yr_built                21534 non-null

In [50]:
#Changing the years data type to int
data['yr_built'] = data['yr_built'].astype(int)
data['yr_renovated'] = data['yr_renovated'].astype(int)
data['Years_Since_Renovation'] = data['Years_Since_Renovation'].astype(int)

#Changing the date column data type to datetime.
data['date'] = pd.to_datetime(data['date'])


In [51]:
#Checking data types after cleaning
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 21534 entries, 0 to 21596
Data columns (total 22 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   id                      21534 non-null  int64         
 1   date                    21534 non-null  datetime64[ns]
 2   price                   21534 non-null  float64       
 3   bedrooms                21534 non-null  int64         
 4   bathrooms               21534 non-null  float64       
 5   sqft_living             21534 non-null  int64         
 6   sqft_lot                21534 non-null  int64         
 7   floors                  21534 non-null  float64       
 8   waterfront              21534 non-null  object        
 9   view                    21534 non-null  object        
 10  condition               21534 non-null  object        
 11  grade                   21534 non-null  object        
 12  sqft_above              21534 non-null  int64 

In [52]:
data.head(30)

id       date      price  bedrooms  bathrooms  sqft_living  \
0   7129300520 2014-10-13   221900.0         3       1.00         1180   
1   6414100192 2014-12-09   538000.0         3       2.25         2570   
2   5631500400 2015-02-25   180000.0         2       1.00          770   
3   2487200875 2014-12-09   604000.0         4       3.00         1960   
4   1954400510 2015-02-18   510000.0         3       2.00         1680   
5   7237550310 2014-05-12  1230000.0         4       4.50         5420   
6   1321400060 2014-06-27   257500.0         3       2.25         1715   
8   2414600126 2015-04-15   229500.0         3       1.00         1780   
9   3793500160 2015-03-12   323000.0         3       2.50         1890   
10  1736800520 2015-04-03   662500.0         3       2.50         3560   
11  9212900260 2014-05-27   468000.0         2       1.00         1160   
12   114101516 2014-05-28   310000.0         3       1.00         1430   
13  6054650070 2014-10-07   400000.0         3       1.75         1370   
14  1175000570 2015-03-12   530000.0         5       2.00         1810   
15  9297300055 2015-01-24   650000.0         4       3.00         2950   
16  1875500060 2014-07-31   395000.0         3       2.00         1890   
17  6865200140 2014-05-29   485000.0         4       1.00         1600   
18    16000397 2014-12-05   189000.0         2       1.00         1200   
19  7983200060 2015-04-24   230000.0         3       1.00         1250   
20  6300500875 2014-05-14   385000.0         4       1.75         1620   
21  2524049179 2014-08-26  2000000.0         3       2.75         3050   
22  7137970340 2014-07-03   285000.0         5       2.50         2270   
23  8091400200 2014-05-16   252700.0         2       1.50         1070   
24  3814700200 2014-11-20   329000.0         3       2.25         2450   
25  1202000200 2014-11-03   233000.0         3       2.00         1710   
26  1794500383 2014-06-26   937000.0         3       1.75         2450   
27  3303700376 2014-12-01   667000.0         3       1.00         1400   
28  5101402488 2014-06-24   438000.0         3       1.75         1520   
29  1873100390 2015-03-02   719000.0         4       2.50         2570   
30  8562750320 2014-11-10   580500.0         3       2.50         2320   

    sqft_lot  floors waterfront       view  ... sqft_above sqft_basement  \
0       5650     1.0         NO       NONE  ...       1180          None   
1       7242     2.0         NO       NONE  ...       2170          None   
2      10000     1.0         NO       NONE  ...        770          None   
3       5000     1.0         NO       NONE  ...       1050          None   
4       8080     1.0         NO       NONE  ...       1680          None   
5     101930     1.0         NO       NONE  ...       3890          None   
6       6819     2.0         NO       NONE  ...       1715          None   
8       7470     1.0         NO       NONE  ...       1050          None   
9       6560     2.0         NO       NONE  ...       1890          None   
10      9796     1.0         NO       NONE  ...       1860          None   
11      6000     1.0         NO       NONE  ...        860          None   
12     19901     1.5         NO       NONE  ...       1430          None   
13      9680     1.0         NO       NONE  ...       1370          None   
14      4850     1.5         NO       NONE  ...       1810          None   
15      5000     2.0         NO       GOOD  ...       1980          None   
16     14040     2.0         NO       NONE  ...       1890          None   
17      4300     1.5         NO       NONE  ...       1600          None   
18      9850     1.0         NO       NONE  ...       1200          None   
19      9774     1.0         NO       NONE  ...       1250          None   
20      4980     1.0         NO       NONE  ...        860          None   
21     44867     1.0         NO  EXCELLENT  ...       2330          None   
22      6300     2.0         NO       NONE  .

#  Simple Linear Regression Modelling

In [53]:
# Import the necessary libraries for statistical modeling
import statsmodels.api as sm
import statsmodels.formula.api as smf

# Assuming 'data' is your DataFrame
# Define the dependent variable 'y' and the independent variable 'x'
y = data['price']
x = data['sqft_living']

# Add a constant term to the independent variable
x = sm.add_constant(x)

# Create a linear regression model using Ordinary Least Squares (OLS)
model = sm.OLS(y, x).fit()

# Print the summary of the regression
print(model.summary())


                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.492
Model:                            OLS   Adj. R-squared:                  0.492
Method:                 Least Squares   F-statistic:                 2.087e+04
Date:                Thu, 21 Dec 2023   Prob (F-statistic):               0.00
Time:                        15:15:32   Log-Likelihood:            -2.9912e+05
No. Observations:               21534   AIC:                         5.982e+05
Df Residuals:                   21532   BIC:                         5.983e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const       -4.215e+04   4404.521     -9.570      

### OLS Regression Summary Explanation

#### Model Fit:
- **R-squared:** 0.492 (49.2% explained variability)
  - The R-squared value indicates that nearly half of the variability in house prices is explained by the model, suggesting a moderate level of predictive power.
- **F-statistic:** 2.087e+04 (p-value: 0.00)
  - The low p-value associated with the F-statistic implies that the overall model is statistically significant, providing evidence that at least one predictor is related to the response variable.

#### Coefficients:
- **Intercept (`const`):** -4.215e+04
  - The intercept represents the estimated house price when the square footage is zero, though in this context, this might not be practically meaningful.
- **sqft_living:** 279.9321
  - The coefficient for 'sqft_living' indicates that, on average, for every one-unit increase in square footage, the predicted house price increases by $279.9321.

#### Hypothesis Testing:
- Coefficient p-values: Very close to zero
  - The very low p-values associated with both the intercept and 'sqft_living' indicate that both coefficients are statistically significant. This means there is strong evidence to reject the null hypothesis that the corresponding coefficients are zero.


#### Conclusion:
In summary, the regression model suggests that the square footage of living space (`sqft_living`) significantly influences house prices (`price`). The model demonstrates a moderate level of explanatory power, and some diagnostics indicate potential issues that warrant further investigation.


## Multiple Linear Regression Modelling.

In [54]:
data.corr()

id     price  bedrooms  bathrooms  sqft_living  \
id                      1.000000 -0.017392  0.000991   0.004971    -0.012381   
price                  -0.017392  1.000000  0.308063   0.525053     0.701587   
bedrooms                0.000991  0.308063  1.000000   0.514368     0.577934   
bathrooms               0.004971  0.525053  0.514368   1.000000     0.755277   
sqft_living            -0.012381  0.701587  0.577934   0.755277     1.000000   
sqft_lot               -0.132061  0.090338  0.032735   0.088921     0.173768   
floors                  0.018573  0.257052  0.177391   0.502285     0.353529   
sqft_above             -0.010967  0.605695  0.479148   0.686173     0.876553   
yr_built                0.021609  0.054273  0.155462   0.507565     0.318217   
yr_renovated           -0.006564  0.051973 -0.001983   0.012100     0.016474   
zipcode                -0.008027 -0.053620 -0.153990  -0.204435    -0.199839   
lat                    -0.002001  0.307868 -0.010415   0.024487     0.052383   
long                    0.020637  0.022417  0.132318   0.225025     0.241856   
sqft_living15          -0.002707  0.585304  0.392874   0.569139     0.756208   
sqft_lot15             -0.138607  0.083189  0.030952   0.088854     0.184801   
Years_Since_Renovation -0.004680  0.110500  0.016407   0.045373     0.041692   

                        sqft_lot    floors  sqft_above  yr_built  \
id                     -0.132061  0.018573   -0.010967  0.021609   
price                   0.090338  0.257052    0.605695  0.054273   
bedrooms                0.032735  0.177391    0.479148  0.155462   
bathrooms               0.088921  0.502285    0.686173  0.507565   
sqft_living             0.173768  0.353529    0.876553  0.318217   
sqft_lot                1.000000 -0.005025    0.184325  0.052646   
floors                 -0.005025  1.000000    0.523367  0.489410   
sqft_above              0.184325  0.523367    1.000000  0.423999   
yr_built                0.052646  0.489410    0.423999  1.000000   
yr_renovated           -0.005873 -0.001946    0.004674 -0.092540   
zipcode                -0.129846 -0.058966   -0.261480 -0.346585   
lat                    -0.085143  0.049419   -0.000889 -0.147850   
long                    0.230137  0.125329    0.345051  0.409666   
sqft_living15           0.145142  0.279176    0.731543  0.326539   
sqft_lot15              0.717095 -0.010850    0.195413  0.070562   
Years_Since_Renovation -0.004715  0.010379    0.007368 -0.228571   

                        yr_renovated   zipcode       lat      long  \
id                         -0.006564 -0.008027 -0.002001  0.020637   
price                       0.051973 -0.053620  0.307868  0.022417   
bedrooms                   -0.001983 -0.153990 -0.010415  0.132318   
bathrooms                   0.012100 -0.204435  0.024487  0.225025   
sqft_living                 0.016474 -0.199839  0.052383  0.241856   
sqft_lot                   -0.005873 -0.129846 -0.085143  0.230137   
floors                     -0.001946 -0.058966  0.049419  0.125329   
sqft_above                  0.004674 -0.261480 -0.000889  0.345051   
yr_built                   -0.092540 -0.346585 -0.147850  0.409666   
yr_renovated                1.000000  0.032844  0.023050 -0.033832   
zipcode                     0.032844  1.000000  0.266595 -0.564279   
lat                         0.023050  0.266595  1.000000 -0.135439   
long                       -0.033832 -0.564279 -0.135439  1.000000   
sqft_living15              -0.007269 -0.279375  0.048569  0.336019   
sqft_lot15                 -0.005177 -0.147519 -0.086035  0.255283   
Years_Since_Renovation      0.340072  0.073451  0.033759 -0.076968   

                        sqft_living15  sqft_lot15  Years_Since_Renovation  
id                          -0.002707   -0.138607               -0.004680  
price                        0.585304    0.083189                0.110500  
bedrooms                     0.392874    0.030952                0.016407  
bathrooms        

In [55]:
import statsmodels.api as sm

# Assuming 'data' is your DataFrame
# Define the dependent variable 'y' and independent variables 'X'
y = data['price']
X = data[['sqft_living', 'bedrooms', 'bathrooms', 'waterfront','view','condition','lat','long','sqft_above','sqft_lot','floors','yr_built','yr_renovated','zipcode','Years_Since_Renovation']]


# Add a constant term to the independent variables
X = sm.add_constant(X)

# Create and fit the multiple linear regression model
model = sm.OLS(y, X).fit()

# Print the summary of the regression
print(model.summary())


ValueError: Pandas data cast to numpy dtype of object. Check input data with np.asarray(data).

#Model Evaluation#

#Recommendation System#

#Conclusion#